packages/renderer-commonmark/src/index.ts

```diff
@@ -103,114 +164,191 @@ export default class CommonmarkRenderer extends Renderer {
   }
 
   /**
-   * Fixed space in Commonmark is the unicode \u00A0 character
+   * A [link](http://commonmark.org) has the url right next to it in Markdown.
    */
-  *'fixed-space'(): IterableIterator<string> {
-    let text = yield;
-    return '\u00A0';
-  },
+  *'link'(props: { href: string, title?: string }): IterableIterator<string> {
+    let [before, text, after] = yield* split();
+    let href = escapeAttribute(props.href);
+    if (props.title) {
+      let title = props.title.replace(/"/g, '\\"');
+      return `${before}[${text}](${href} "${title}")${after}`;
+    }
+    return `${before}[${text}](${href})${after}`;
+  }
 
   /**
-   * A [link](http://commonmark.org) has the url right next to it in Markdown.
+   * A `code` span can be inline or as a block:
+   *
+   * ```js
+   * function () {}
+   * ```
    */
-  *'link'(props: { href: string }): IterableIterator<string> {
-    let [before, text, after] = yield* split();
-    return `${before}[${text}](${props.href})${after}`;
+  *'code'(props: { style: CodeStyle, info?: string }, state: State): IterableIterator<string> {
+    state.push({ isPreformatted: true, htmlSafe: false });
+    let text = yield;
+    state.pop();
+    let code = text.join('');
+
+    if (props.style === 'fence') {
+      code = '\n' + code;
+      let info = props.info || '';
+      let newlines = '\n';
+      if (state.get('isList') && state.get('nextAnnotation')) {
+        newlines += '\n';
+      }
+
+      if (code.indexOf('```') !== -1) {
+        return `~~~${info}${code}~~~${newlines}`;
+      } else {
+        return `\`\`\`${info}${code}\`\`\`${newlines}`;
```

Comment:
Yes, since all code blocks end with a newline. It's weird. I could do some detection to add a newline if there isn't one, but I don't think it's super worthwhile